### PR TITLE
fix: misc fixes required to load from genesis

### PR DESCRIPTION
**Pull request by arjain4**

### CODE REVIEW

**Review by ripatel-fd — CHANGES_REQUESTED** [review]
https://github.com/firedancer-io/firedancer/actions/runs/10435968296/job/28900463377?pr=2737
Bank hash mismatch

File: src/flamenco/runtime/fd_runtime.c [hunk 1]
```diff
@@ -70,8 +70,8 @@ fd_runtime_init_bank_from_genesis( fd_exec_slot_ctx_t *  slot_ctx,
   memset(slot_ctx->slot_bank.banks_hash.hash, 0, FD_SHA256_HASH_SZ);
 
   slot_ctx->slot_bank.fee_rate_governor = genesis_block->fee_rate_governor;
-  slot_ctx->slot_bank.lamports_per_signature = 5000UL;
-  slot_ctx->prev_lamports_per_signature = 5000UL;
+  slot_ctx->slot_bank.lamports_per_signature = 0UL;
+  slot_ctx->prev_lamports_per_signature = 0UL;
```
Comment (jumpsiegel) [inline src/flamenco/runtime/fd_runtime.c:75]: This is a surprising change

File: src/flamenco/runtime/fd_runtime.c [hunk 2]
```diff
@@ -100,14 +100,14 @@ fd_runtime_init_bank_from_genesis( fd_exec_slot_ctx_t *  slot_ctx,
...
+  elem->fee_calculator.lamports_per_signature = 0UL;
 
   slot_ctx->slot_bank.block_hash_queue.ages_root = NULL;
   slot_ctx->slot_bank.block_hash_queue.ages_pool = fd_hash_hash_age_pair_t_map_alloc( slot_ctx->valloc, 400 );
   fd_hash_hash_age_pair_t_mapnode_t * node = fd_hash_hash_age_pair_t_map_acquire( slot_ctx->slot_bank.block_hash_queue.ages_pool );
   node->elem = (fd_hash_hash_age_pair_t){
     .key = *genesis_hash,
-    .val = (fd_hash_age_t){ .hash_index = 0, .fee_calculator = (fd_fee_calculator_t){.lamports_per_signature = 5000}, .timestamp = (ulong)fd_log_wallclock() }
+    .val = (fd_hash_age_t){ .hash_index = 0, .fee_calculator = (fd_fee_calculator_t){.lamports_per_signature = 0UL}, .timestamp = (ulong)fd_log_wallclock() }
   };
```
[the remark this answers is not in the document]
Comment (arjain4) [reply src/flamenco/runtime/fd_runtime.c:111]: that is correct, agave sets this value to 0

**Comment by arjain4** [comment]
> https://github.com/firedancer-io/firedancer/actions/runs/10435968296/job/28900463377?pr=2737 Bank hash mismatch

That ledger test uses a version of agave which has a bug related to the fees sysvar. Removing the test

**Pull request merged**